plugins/telemetry-otel/src/internalClusterTest/java/org/opensearch/telemetry/tracing/TelemetryTracerDisabledSanityIT.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.telemetry.tracing;
+
+import org.opensearch.client.Client;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.plugins.Plugin;
+import org.opensearch.telemetry.OTelTelemetrySettings;
+import org.opensearch.telemetry.TelemetrySettings;
+import org.opensearch.test.OpenSearchIntegTestCase;
+
+import static org.opensearch.index.query.QueryBuilders.queryStringQuery;
+
+@OpenSearchIntegTestCase.ClusterScope(scope = OpenSearchIntegTestCase.Scope.TEST, supportsDedicatedMasters = false, minNumDataNodes = 2)
+public class TelemetryTracerDisabledSanityIT extends OpenSearchIntegTestCase {
```

Comment:
I think these tests should be moved to server module, under `internalClusterTest`

Reply:
These tests are dependent on OTel plugin, thats why added it in plugin itself. Other plugins also have their specific integ test cases.

Also all other tests (Server included) use MockTelemetryPlugin for emitting spans. I think keeping it in plugins will have proper segregation.

Reply:
I think the `MockTelemetryPlugin` (which is server actually) is a temporary measure, the dependency goes the other way - server brings the plugin, not plugin brings the server.